### DISCUSSION TITLE
How to End-to-End Performance testing a typical implementation?

### DISCUSSION THREAD

**Discussion by vsmbee** [ory/hydra#3287] [Q&A]
I think we have a fairly common case with Ory Hydra, PostgreSQL and Login/Consent app implemented in Java, both services deployed in a bare metal k8s cluster.
Is there a simple and reasonably relevant way to do performance testing of the complete installation? I'm currently trying to replace front-end part with a mock (another java service) but _oauth2_authentication_session_ is eluding me and I can not complete the authorization code flow.
Is there some browser-like E2E testing framework or some best practice?

**Comment by vinckr** [comment]
Hey vsmbee, 
Take a look at the tests we use for our examples: https://github.com/ory/docs/tree/master/tests as well as the ones we use to test Hydra: https://github.com/ory/hydra/tree/master/test 
So we use [Playwright](https://playwright.dev/) and [Cypress](https://www.cypress.io/) but there are many other options probably.
